synapse/util/iterutils.py

```diff
@@ -135,3 +135,54 @@ def sorted_topologically(
                 degree_map[edge] -= 1
                 if degree_map[edge] == 0:
                     heapq.heappush(zero_degree, edge)
+
+
+def sorted_topologically_batched(
```

Comment:
FWIW I think `sorted_topologically` could become

```python
def sorted_topologically(nodes, graph):
	for batch in sorted_topologically_batched(nodes, graph):
        yield from batch
```

It would probably create a bunch of intermediate lists and such though, which is unnecessary.

Reply:
Yes, ish. I think `sorted_topologically` gives a lexicographical topological sort, which could return a slightly different order (though I don't think we really rely on that fact anywhere, so)